jwst/ramp_fitting/ramp_fit.py

```diff
@@ -2940,12 +2940,12 @@ def calc_num_seg(gdq, n_int):
     """
     max_cr = 0  # max number of CRS for all integrations
 
-    # For all 2d pixels, get max number of CRs along their ramps
-    # to use as a surrogate for the number of semiramps along the ramps
+    # For all 2d pixels, get max number of CRs or DO_NOT_USE flags along their
+    # ramps, to use as a surrogate for the number of semiramps along the ramps
     for nint in range(n_int):  # loop over integrations
-        gdq_cr = np.bitwise_and( gdq[nint, :, :, :], dqflags.group['JUMP_DET'])
-        temp_max_cr = int((gdq_cr.sum(axis=0)).max()/dqflags.group['JUMP_DET'])
-        max_cr = max( max_cr, temp_max_cr )
+        gdq_cr = np.bitwise_and(gdq[nint], dqflags.group['JUMP_DET'] | dqflags.group['DO_NOT_USE'])
```

Comment:
The expression:

```python
dqflags.group['JUMP_DET'] | dqflags.group['DO_NOT_USE']
```
always evaluates to 5.  So the above is really:

```python
gdq_cr = np.bitwise_and(gdq[nint], 5)
```
which will pick out all the 1, 4 and 5 values in `gdq`.  For example:

```
In [24]: gdq                                                                                                          
Out[24]: array([0, 1, 4, 5, 8])

In [25]: np.bitwise_and(gdq, 5)                                                                                       
Out[25]: array([0, 1, 4, 5, 0])
```

But then when you sum these, the 4 and 5 are going to give you a much higher count.  Instead of summing you may want to count them using `np.count_nonzero`.  Summing was fine when the values were all 4 and that could be corrected by dividing by 4, but not now.

Reply:
Also, please write a quick unit test for this so we don't break it again.  😄

Reply:
Hmmm, could've sworn I had put in an appropriate expression that basically returned an array of 1's and 0's (equivalent to True/False), such that the .sum() really did just count up the number of hits, as opposed to the sum of all the flag values. Will look at that again.

Unit tests: to quote the Beatles, "yeah, yeah, yeah, ..."

Reply:
Also also, I would pick either:

```python
np.bitwise_and(gdq[nint], np.bitwise_or(dqflags.group['JUMP_DET'], dqflags.group['DO_NOT_USE']))
```

or 

```python
gdq[int] & (dqflags.group['JUMP_DET'] | dqflags.group['DO_NOT_USE'])
```

not mix up the function calls and operators (which point to the function calls).

Also also also, might be nice to put something like

```python
DO_NOT_USE = dqflags.group['DO_NOT_USE']
```

globally in this module as we've done elsewhere to make these sorts of things a bit more clear and concise:

```python
gdq[int] & (JUMP_DET | DO_NOT_USE)
```